addons/mail/static/src/models/thread/thread.js

```diff
@@ -1385,8 +1388,10 @@ function factory(dependencies) {
             if (this.channel_type === 'chat' && this.correspondent) {
                 return this.custom_channel_name || this.correspondent.nameOrDisplayName;
             }
-            if (this.channel_type === 'group') {
-                return this.name || this.members.map(partner => partner.nameOrDisplayName).join(', ');
+            if (this.channel_type === 'group' && !this.name) {
+                const partnerNames = this.members.map(partner => partner.nameOrDisplayName).sort();
+                const guestNames = this.guestMembers.map(guest => guest.name).sort();
```

Comment:
Ideally we should use Collator for this `sort` (task https://www.odoo.com/web#id=2780165&menu_id=4720&cids=1&action=333&active_id=1519&model=project.task&view_type=form) (not on this PR).

But until then it would be nice to at least compare with case-insensitive, like we do in discuss sidebar.